Sync `css/css-properties-values-api` from WPT upstream

Ahmad-S792 opened the pull request:
#### 984de83064446098ffd0ed2a896b8e0a48cffed5
<pre>
Sync `css/css-properties-values-api` from WPT upstream

<a href="https://bugs.webkit.org/show_bug.cgi?id=293180">https://bugs.webkit.org/show_bug.cgi?id=293180</a>

Reviewed by Tim Nguyen.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/d2b8b72ee2b97f942fa12d5fcdcf6c01ff38618a">https://github.com/web-platform-tests/wpt/commit/d2b8b72ee2b97f942fa12d5fcdcf6c01ff38618a</a>

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/custom-property-animation-used-in-shorthand.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/custom-property-transition-length-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/custom-property-transition-length.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/custom-property-transition-number-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/custom-property-transition-number.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-animation.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-shadow-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-shadow.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/register-property-syntax-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/register-property-syntax-parsing.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/resources/utils.js:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/unit-cycles-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/unit-cycles.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/var-reference-registered-properties-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/var-reference-registered-properties.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/custom-property-transition-number-percentage-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/custom-property-transition-number-percentage.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/registered-neutral-keyframe-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/registered-neutral-keyframe.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/registered-var-to-registered-animating-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/registered-var-to-registered-animating.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/crashtests/computed-property-universal-syntax.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/crashtests/consume-color-contrast-crash.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/crashtests/initial-in-audio-crash.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/crashtests/transition-to-none-crash-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/crashtests/transition-to-none-crash-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/crashtests/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/initial-value-unit-arithmetic-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/initial-value-unit-arithmetic.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/non-computed-unit-cycles-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/non-computed-unit-cycles.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/register-property-sign-mixed-lengths-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/register-property-sign-mixed-lengths.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-001-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-001-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-002-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-003-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-003-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-003.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-004-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-004.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-005-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-005.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-parse-crash.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/resolved-color-value-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/resolved-color-value.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/var-reference-registered-properties-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/var-reference-registered-properties-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/WEB_FEATURES.yml:

Canonical link: <a href="https://commits.webkit.org/295070@main">https://commits.webkit.org/295070@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/13dbc7712d86e4ad54af2bc7b8d52c12a4440578

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/103963 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/23666 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/13987 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/109169 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/54626 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/24025 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/32211 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/78982 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/106969 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/18657 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/93787 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/59310 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/18447 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/11838 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/53991 "Built successfully") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/88190 "Found 1 new API test failure: TestWebKitAPI.CARingBufferTest.FetchTimeBoundsConsistent (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/11896 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/111543 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/31119 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/22931 "Found 1 new test failure: imported/w3c/web-platform-tests/html/semantics/embedded-content/media-elements/loading-the-media-resource/resource-selection-invoke-remove-from-document-networkState.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/87989 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/31483 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/89987 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/87645 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22315 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/32530 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/10274 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/25503 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/31048 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/30841 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/34178 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/32402 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/register-property-syntax-parsing.html:
```diff
@@ -49,7 +36,6 @@
 assert_valid("<length>", "calc(2px*4 + 10px)");
 assert_valid("<length>", "7.1e-4cm");
 assert_valid("<length>", "calc(7in - 12px)");
-assert_valid("<length>", "calc(15px + (sign(100vh - 10px) * 5px))");
```
This is undoing some of @weinig's unexported changes in https://github.com/WebKit/WebKit/commit/b59ab1a6d619275fa32c3ee570f7b4cf0d97cd3d

Ahmad-S792 replied:
@nt1m - Those tests were split by Emilio here - https://github.com/web-platform-tests/wpt/pull/49801

Which we have in this PR as well, so we are not losing any coverage.

Commit:
Sync `css/css-properties-values-api` from WPT upstream

https://bugs.webkit.org/show_bug.cgi?id=293180

Reviewed by Tim Nguyen.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/d2b8b72ee2b97f942fa12d5fcdcf6c01ff38618a

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/custom-property-animation-used-in-shorthand.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/custom-property-transition-length-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/custom-property-transition-length.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/custom-property-transition-number-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/custom-property-transition-number.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-animation.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-shadow-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-shadow.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/register-property-syntax-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/register-property-syntax-parsing.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/resources/utils.js:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/unit-cycles-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/unit-cycles.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/var-reference-registered-properties-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/var-reference-registered-properties.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/custom-property-transition-number-percentage-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/custom-property-transition-number-percentage.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/registered-neutral-keyframe-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/registered-neutral-keyframe.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/registered-var-to-registered-animating-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/registered-var-to-registered-animating.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/animation/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/crashtests/computed-property-universal-syntax.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/crashtests/consume-color-contrast-crash.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/crashtests/initial-in-audio-crash.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/crashtests/transition-to-none-crash-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/crashtests/transition-to-none-crash-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/crashtests/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/initial-value-unit-arithmetic-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/initial-value-unit-arithmetic.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/non-computed-unit-cycles-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/non-computed-unit-cycles.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/register-property-sign-mixed-lengths-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/register-property-sign-mixed-lengths.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-001-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-001-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-002-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-003-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-003-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-003.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-004-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-004.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-005-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-005.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-parse-crash.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/resolved-color-value-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/resolved-color-value.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/var-reference-registered-properties-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/var-reference-registered-properties-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/WEB_FEATURES.yml:

Canonical link: https://commits.webkit.org/295070@main
